framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -974,18 +960,20 @@ public void checkMethod(MethodTree methodDeclTree, AnnotatedExecutableType type)
      * @param methodType annotated type of the method
      */
     private void checkTypeVariables(MethodTree node, AnnotatedExecutableType methodType) {
-        JavaExpressionContext context =
-                JavaExpressionContext.buildContextForMethodDeclaration(node, factory.getChecker());
         for (int i = 0; i < methodType.getTypeVariables().size(); i++) {
             AnnotatedTypeMirror atm = methodType.getTypeVariables().get(i);
-            viewpointAdaptToContext(context, atm);
+            convertAnnotatedTypeMirror(
+                    stringExpr ->
+                            StringToJavaExpression.atMethodBody(
+                                    stringExpr, node, factory.getChecker()),
+                    atm);
             checkType(atm, node.getTypeParameters().get(i));
         }
     }
 
     /**
-     * Returns true if {@code am} is an expression annotation, that is an annotation whose value is
-     * a Java expression.
+     * Returns true if {@code am} is an expression annotation, that is an annotation whose element
+     * is a Java expression.
      *
      * @param am an annotation
      * @return true if {@code am} is an expression annotation
```

Comment:
This method could be made much more efficient by looking up in the map rather than iterating over it.  (Or create another map or set of the string names and look up in that.)

Reply:
This can be fixed when the type of `annoToElements` is changed. (See the TODO at the declaration of `annoToElements`.)